packages/shopping/src/spec-enhancers/info-spec.extension.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: loopback4-example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {bind, Context, inject} from '@loopback/context';
+import {
+  asSpecEnhancer,
+  mergeOpenAPISpec,
+  OASEnhancer,
+  OpenApiSpec,
+} from '@loopback/openapi-v3';
+import {PackageInfo, PackageKey} from '../application';
```

Comment:
Due to the order of `require`, `PackageKey` is `undefined` when this file is loaded. As a result, `@inject` fails because the binding key is `undefined`.

Moving `PackgeKey` from `application.ts` to `keys.ts` fixes the issue.